topics/lmms/rqtl2/gn-rqtl-design-implementation.gmi

```diff
@@ -0,0 +1,118 @@
+# RQTL Implementation for GeneNetwork Design Proposal
+
+## Tags
+
+* Assigned: alexm,
+* Keywords: RQTL, GeneNetwork2, Design
+* Type: Enhancements,
+* Status: In Progress
+
+
+
+## Description
+
+This document outlines the design proposal for the re-implementation of the RQTL feature in GeneNetwork providing also  a console view to track the external process.
+
+### Problem Definition
+The current RQTL implementation requires enhancements. The core functionality remains the same: making API calls from GeneNetwork 3 (GN3). The system needs a cleaner architecture for future use, emphasizing improved error handling and a clear separation of concerns between GeneNetwork 2 (GN2) and GN3. This will involve eliminating file transfers between GN2 and GN3. Additionally, a console should be provided to users for tracking the progress of ongoing tasks.
+
+## High-Level Design
+This is divided into two major components:
+* RQTL Api
+
+* Monitoring system for the rqtl script
+
+### RQTL Api 
+This component serves as the entry point to the gn3 rqtl system and can be subdivided into
+data validation and preprocessing stage, data computation stage, output processing stage.
+In the component the task mainly   involves rewriting the current API to replace file transfers between GN2 and GN3 with JSON data We will also implement tests and enhance error handling.
+
+
+**Subtasks:**
+
+- [ ] Rewrite the RQTL API endpoints
+- [ ] add validation and preprocessing  for data submitted from the 
+- [ ]  Processing output from  the external script
+- [ ] Add unit tests for this module
+- [ ] Implement better error handling for the API
+
+###  Monitoring system for the  rqtl script
+
+This component involves creating a monitoring system to track the state of the external process and output relevant information to the user.
```

Comment:
Some questions:

1. what are the states of an external process? I could suggest `QUEUED, STARTED, INPROGRESS, COMPLETED`
2. what do we define as an external process? i.e. is this the full run of RQTL or one of the subdivided steps
3. what relevant information do we want? i.e. processing times per stage, memory/cpu use, etc.

Reply:
this commit addresses this issue: https://github.com/genenetwork/gn-gemtext-threads/pull/46/commits/2db603e7a0848e5792b930d3e4731728fcfba8e9